shared-services/README.md

```diff
@@ -0,0 +1,172 @@
+
+If this is a new repository, you should use the flake template.
+
+```
+nix flake init --template github:zeme-iohk/devx
+```
+
+Two files will be created in the top-level: `flake.nix` and `devx.nix`.
+
+[Click here](#the-devx-flakenix-file) to continue learning.
+
+# Integration with an existing repository
+
+In order to successully migrate to `devx` your nix code will have to be refactored until your `flake.nix` is standardized (see [here](#the-standard-flakenix-file)) and your `devx.nix` file (which you have to create) becomes the entrypoint to all your nix code (see [here](#the-devxnix-file)).
+
+# The standard `flake.nix` file
+
+Projects using `devx` share an almost identical `flake.nix`:
+```nix
+{
+  description = "";
+
+  inputs.devx.url = "github:zeme-iohk/devx";
+
+  # You can add other flake inputs here if you must: inputs.other = {}
+
+  outputs = inputs.devx.lib.mkFlakeOutputs;
+
+  nixConfig = {
+    extra-substituters = [
+      "https://cache.iog.io"
+      "https://cache.zw3rk.com"
+    ];
+    extra-trusted-public-keys = [
+      "hydra.iohk.io:f/Ea+s+dFdN+3Y/G+FDgSq+a5NEWhJGzdjvKNGv0/EQ="
+      "loony-tools:pr9m4BkM/5/eSTZlkQyRt57Jz7OMBxNSUiMC4FkcNfk="
+    ];
+    accept-flake-config = true;
+    allow-import-from-derivation = true;
+  };
+}
+```
+You may edit this file in three ways:
+1. Fill the `description` to your liking.
+2. Edit the `nixConfig` attrset, but do not **no** invalidate the cache-related fields.
+3. Add new inputs to the flake. *Note that `devx` provisions and manages all other flake inputs, and therefore should be the only input in your flake. You may nonetheless declare additional inputs as you would normally, and they will be available in `devx.nix`.*
+
+# The `devx.nix` file
+
+The `devx.nix` file is the entrypoint to all your nix code:
+```nix
+devx: 
+{
+  supportedSystems = ["x86_64-linux" "x86_64-darwin"]; 
+  
+  ghcVersions = ["ghc8107" "ghc901" "ghc924"]; 
+
+  haskellNixProject = ./haskell.nix;
+
+  extraFlakeOutputs = {};
+
+  shellHooks = {
+    default = {
+      hook1-name = ''bash code'';
+      hook2-name = ''bash code'';
+    };
+    shell1 = {};
+    shell2 = {};
+  };
+
+  shellCommands = {
+    default = {
+      command1-name = {
+        description = "";
+        category = "";
+        exec = "bash code";
+      };
+      command2-name = {
+        description = "";
+        category = "";
+        exec = "bash code";
+      };
+    }];
+    shell1 = {};
+    shell2 = {};
+  };
+}
+```
+    
+- `supportedSystems` should be any of "x86_64-linux" "x86_64-darwin" "aarch64-linux" "aarch64-darwin".
+  Flake outputs for packages and devshells will be produced for each supported system.
+    
+- `ghcVersions` should be any of "ghc8106" "ghc901" "ghc924". This will determine the version of the haskell toolchain. A devShell will be avaiable for each GHC version.
+
+- `haskellNixProject` path to a standard `./haskell.nix` file. Click [here](#the-haskellnix-file) for reference. 
```

Comment:
What does this file do? Or did you mean this to be cabal.project? If so, would be better by default to just get files like this from known fs locations (which are accessible via self in the inputs passed to the flake function)